numpy/lib/nanfunctions.py

```diff
@@ -214,7 +216,11 @@ def _divide_by_count(a, b, out=None):
                 return np.divide(a, b, out=out, casting='unsafe')
         else:
             if out is None:
-                return a.dtype.type(a / b)
+                # Precaution against reduced object arrays
+                try:
+                    return a.dtype.type(a / b)
+                except AttributeError:
```

Comment:
Is this faster than checking for the attribute as you do below?

Reply:
During a quick local test it shaves off about 100 ns (~400 vs ~300 ns) for the non-object case.
Would you feel this performance gain is worthwhile enough to stick to the current `try`/`except` block?

Reply:
I can't see it making a difference in the big picture.  There might be a small difference in clarity, but with one liners that is debatable ...